lib/liquid/ast.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Liquid.Ast do
+  @moduledoc """
+  Builds the AST processing with Nimble, only liquid valid tags and variables. It uses Tokenizer
+  to send to Nimble only tags and variables, without literals.
+  Literals (any markup which is not liquid variable or tag) are slow to be processed by Nimble thus
+  this module improve performance between 30% and 100% depending how much text is processed.
+  """
+  alias Liquid.{Tokenizer, Parser, Block}
+
+  @doc """
+  Recursively builds the AST taking a markup, or a tuple with a literal and a rest markup.
+  It uses `context` to validate the correct opening and close of blocks and sub blocks.
+  """
+  @spec build(String.t() | {String.t(), String.t()}, Keyword.t(), List.t()) ::
```

Comment:
Can you go with a single style of type, preferably shorthands? In this case:

```
@spec build(binary | {binary, binary}, Keyword.t(), list) :: {:ok, list, Keyword.t(), binary} | {:error, binary, binary}
```

Thanks :)

Reply:
Done